### PR TITLE
introduce transform parser [as part of onging evaluation]

### PR DESCRIPTION
**Pull request by chenqin**
aim to support primitive passthrough tranform input data stream

- cast from timestamp to unix_time
- add column based on compute value of multiple row
- filter or downsample based on row value

instead of creating own yaml configuration, use subset of ansi SQL to and on demand transform lambda function

https://druid.apache.org/docs//0.15.1-incubating/ingestion/transform-spec.html

### CODE REVIEW

File: ext/libpg_query_Ext.cmake [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# not sure why we only add this for APPLE only.
```
Comment (shawncao) [inline ext/libpg_query_Ext.cmake:1]: why? :)

File: CMakeLists.txt [hunk 2]
```diff
@@ -193,6 +193,8 @@ find_package(Boost 1.65 MODULE
     thread
   REQUIRED)
 
+include(ext/libpg_query_Ext.cmake)
```
Comment (shawncao) [inline CMakeLists.txt:196]: To be naming consistent - could we rename this file as `PGQuery_Ext` or `PG_Ext`?

File: src/ingest/test/TestIngestion.cpp [hunk 3]
```diff
@@ -73,6 +74,17 @@ TEST(IngestTest, TestSpecGeneration) {
 #endif
 }
 
+TEST(IngestTest, TestTransformerAddColumn) {
+  //schema: "ROW<id:int, event:string, items:list<string>, flag:bool, value:tinyint>"
+  auto transformer = pg_query_parse("SELECT id, event, items, flag, value, to_unixtime(now) from nebula.test");
+
+  printf("%s\n", transformer.parse_tree);
```
Comment (shawncao) [inline src/ingest/test/TestIngestion.cpp:81]: nit: `LOG(INFO)<<` instead?

**Review by shawncao — APPROVED** [review]
LGTM. Could you please rename the ext file to be consistent with others?

File: ext/libpg_query_Ext.cmake (hunk 1 above)
Comment (chenqin) [reply ext/libpg_query_Ext.cmake:1]: removed.

File: CMakeLists.txt (hunk 2 above)
Comment (chenqin) [reply CMakeLists.txt:196]: fixed.

File: src/ingest/test/TestIngestion.cpp (hunk 3 above)
Comment (chenqin) [reply src/ingest/test/TestIngestion.cpp]: fixed.

**Comment by chenqin** [comment]
The goal of this pr is to evaluate sql / js approach side by side. If JS approach wins, this dep will be removed.

**Pull request merged**